lib/cylc/task_proxy.py

```diff
@@ -659,33 +781,154 @@ def handle_event(
+        conf = self.tdef.rtconfig["events"]
+        if (key in self.event_handler_try_states
+                or event not in conf["mail events"]):
+            return
+
+        names = [self.suite_name, str(self.point), self.tdef.name]
+        if self.submit_num:
+            names.append("%02d" % self.submit_num)
+        subject = "[%(event)s] %(names)s" % {
+            "event": event, "names": ".".join(names)}
+        cmd = ["mail", "-s", subject]
+        # From:
+        if conf["mail from"]:
+            cmd += ["-r", conf["mail from"]]
+        else:
+            cmd += ["-r", "notifications@" + get_suite_host()]
+        # To:
+        if conf["mail to"]:
+            cmd.append(conf["mail to"])
+        else:
+            cmd.append(user)
+        # Mail message
+        stdin_str = "%s: %s\n" % (subject, message)
+        # SMTP server
+        env = None
+        if conf["mail smtp"]:
+            env = dict(os.environ)
+            env["smtp"] = conf["mail smtp"]
+
+        ctx = SuiteProcContext(key, cmd, env=env, stdin_str=stdin_str)
+        try_state = TryState(ctx, conf["mail delays"])
+        try_state.next()
+        self.event_handler_try_states[key] = try_state
+        if try_state.next() is None or try_state.is_delay_done():
+            try_state.timeout = None
+            SuiteProcPool.get_inst().put_command(
+                ctx, self.event_handler_callback)
+        else:
+            self.log(
+                INFO,
+                "%s will run after %s (after %s):\n\t%s" % (
+                    self.EVENT_MAIL,
+                    try_state.delay_as_seconds(),
+                    try_state.timeout_as_str(),
+                    ctx.cmd))
+
+    def call_event_handlers(self, event, message, only_list=None):
+        """Call custom event handlers."""
+        conf = self.tdef.rtconfig["events"]
+        handlers = []
+        if self.event_hooks[event + ' handler']:
+            handlers = self.event_hooks[event + ' handler']
+        elif conf["handler"] and event in conf['handler events']:
+            handlers = self.tdef.rtconfig["events"]["handler"]
+        env = None
+        for i, handler in enumerate(handlers):
+            key = (
+                "%s-%02d" % (self.EVENT_HANDLER, i),
+                event,
+                "%02d" % self.submit_num)
+            if key in self.event_handler_try_states or (
+                    only_list and i not in only_list):
+                continue
+            cmd = handler % {
+                "event": quote(event),
+                "suite": quote(self.suite_name),
+                "point": quote(str(self.point)),
+                "name": quote(self.tdef.name),
+                "submit_num": self.submit_num,
+                "id": quote(self.identity),
+                "message": quote(message),
+            }
+            if cmd == handler:
+                # Nothing substituted, assume classic interface
```

Comment:
I guess the user will have to continue to do what they need to do now:

```cfg
failed handler = echo 'only this' # 
```

The command line is then:

```bash
echo 'only this' # event suite task.cycle message
```